### PR TITLE
Closes #247 247_update_deprecation: update programming strategy

### LINKED ISSUE #247: Documentation: Update Deprecation Strategy
### Please select a category the issue is focused on?

Developer Guides

### Let us know where something needs a refresh or put your idea here!

The "Deprecation" section in the programming strategy should be updated:

- The unit tests of a deprecated function should be kept in phase 1 (warning). The call of the deprecated function needs to be enclosed in `suppress_warnings()` (example code should be added). In phase 2 (error) the unit tests should be removed (except the test for checking the deprecation message).
- The unit test for checking the deprecation message should be added to the original test file and not to `test-deprecation.R`.

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/programming_strategy.Rmd [hunk 1]
```diff
@@ -683,50 +695,61 @@ parameters which are not passed as `exprs()` argument (e.g. `new_var = VAR1` or
...
-Unit tests for deprecated functions and parameters must be added to
-`tests/testthat/test-deprecation.R` to ensure that a warning or error is issued. 
+Unit tests for deprecated functions and arguments must be added to the test
+file of the function to ensure that a warning or error is issued.
 
-When writing the unit test, check that the error or warning has the right class, i.e. "lifecycle_error_deprecated" or "lifecycle_warning_deprecated", respectively. The unit-test should follow the corresponding format, per the [unit test guidance](unit_test_guidance.html#writing-unit-tests-in-admiral-):
+When writing the unit test, check that the error or warning has the right class,
+i.e. `"lifecycle_error_deprecated"` or `"lifecycle_warning_deprecated"`,
+respectively. The unit-test should follow the corresponding format, per the
+[unit test guidance](unit_test_guidance.html#writing-unit-tests-in-admiral-).
```
Comment (bms63) [inline vignettes/programming_strategy.Rmd:704]: Is the hyphen at the end okay?  - guess it clear the link checker

File: vignettes/programming_strategy.Rmd [hunk 2]
```diff
@@ -683,50 +695,61 @@ parameters which are not passed as `exprs()` argument (e.g. `new_var = VAR1` or
 
 ## Unit Testing
 
-Unit tests for deprecated functions and parameters must be added to
-`tests/testthat/test-deprecation.R` to ensure that a warning or error is issued. 
+Unit tests for deprecated functions and arguments must be added to the test
+file of the function to ensure that a warning or error is issued.
```
Comment (bms63) [inline vignettes/programming_strategy.Rmd:699]: So the file `test-deprecation.R` will be removed at some point?

Can you provide a short example?  For example,  The function derive_var_example() is located in the `example.R` file.  The unit test will be found in `tests/testthat/test-example.R` file and should be updated there????

Comment (bundfussr) [reply vignettes/programming_strategy.Rmd]: Yes, in the Q3 release `test-deprecation.R` should be removed.

I have added a footnote regarding the naming of the test file.

File: vignettes/programming_strategy.Rmd (hunk 1 above)
Comment (bundfussr) [reply vignettes/programming_strategy.Rmd]: Good catch! The hyphen breaks the link partially. I have fixed it.

File: vignettes/programming_strategy.Rmd (hunk 2 above)
Comment (bundfussr) [reply vignettes/programming_strategy.Rmd]: I don't think we need a separate issue for it. It should be covered by the general deprecation process issue for Q3.

**Review by bms63 — APPROVED** [review]
Looks good to me!
